src/cargo/util/toml_mut/dependency.rs

```diff
@@ -921,6 +921,47 @@ impl Display for WorkspaceSource {
     }
 }
 
+impl Dependency {
+    /// Convert denpency as feature to TOML.
+    /// Panics if the path is relative
+    pub fn to_toml_as_feature(&self, crate_root: &Path) -> toml_edit::Item {
+        assert!(
+            crate_root.is_absolute(),
+            "Absolute path needed, got: {}",
+            crate_root.display()
+        );
+        let features: toml_edit::Value = vec![self.as_feature()].iter().collect();
+        toml_edit::value(features)
+    }
+
+    pub fn as_feature(&self) -> String {
+        if let Some(rename) = &self.rename {
+            format!("dep:{}", rename)
+        } else {
+            format!("dep:{}", self.name)
+        }
+    }
+
+    /// Check whether `dep:<dep>` is defined in the features section.
+    ///
+    /// `true`  if there is a define `dep:<dep>`
+    /// `false` if `dep:<dep>` is not used ever.
+    ///
+    /// Make sure that `dep:<dep>` is included in one of features in the features table.
+    pub fn is_active_as_feature(&self, item: &mut toml_edit::Item) -> bool {
```

Comment:
By making this `pub` on `Dependency`, this over-generalizes some fairly specific logic.

Seems like this would be better as a free function near the lone caller

Reply:
Resolved and use `Manifest::is_explicit_dep_activation` instead